lib/src/tasks/analyze/cli.dart

```diff
@@ -41,10 +43,18 @@ class AnalyzeCli extends TaskCli {
     bool fatalWarnings = TaskCli.valueOf(
         'fatal-warnings', parsedArgs, config.analyze.fatalWarnings);
     bool hints = TaskCli.valueOf('hints', parsedArgs, config.analyze.hints);
+    bool fatalHints = TaskCli.valueOf('fatal-hints', parsedArgs, config.analyze.failOnHints);
+
+    if (!hints && fatalHints) {
+      return new CliResult.fail('You must enable hints to fail on hints.');
+    }
 
     AnalyzeTask task = analyze(
-        entryPoints: entryPoints, fatalWarnings: fatalWarnings, hints: hints);
-    reporter.logGroup(task.analyzerCommand, outputStream: task.analyzerOutput);
+        entryPoints: entryPoints, fatalWarnings: fatalWarnings, hints: hints, fatalHints: fatalHints);
+    var title = task.analyzerCommand;
+    if (fatalHints) title += ' (treating hints as fatal)';
```

Comment:
since fatal hints aren't an option for dartanalyzer (yet) I just wanted to tack this on to the output to differentiate it